Sources/General/KingfisherOptionsInfo.swift

```diff
@@ -203,6 +203,10 @@ public enum KingfisherOptionsInfoItem {
     /// value to overwrite the config setting for this caching item.
     case memoryCacheExpiration(StorageExpiration)
     
+    /// By default, every read operation for an item is extending it's expiration time. Use this setting to
+    /// disable this behavior and expire item regardless it's next requests.
+    case memoryCacheExpirationNotExtendable
```

Comment:
It's good to have this option. However, maybe we could make it more generic. Currently, we are extending the expiration duration according to the original `.memoryCacheExpiration` setting. Let's make it something like `.memoryCacheAccessExtendingExpiration(StorageExpiration)` so we can control it better. For example, in your case, you send a `. memoryCacheAccessExtendingExpiration(.seconds(0))` to prevent the extending, while others can choose `. memoryCacheAccessExtendingExpiration(.days(3))` to extend it by 3 days or `.memoryCacheAccessExtendingExpiration(.expired)` to make it expired immediately.

And maybe we could do the same thing to the disk cache, so we can solve #1152 as well.

Reply:
@onevcat Please take a look at the implementation of `.memoryCacheAccessExtendingExpiration(StorageExpiration)`.

In case of not extending I use `.memoryCacheAccessExtendingExpiration(.never)`.

Reply:
@dstranz Great!

> In case of not extending I use .memoryCacheAccessExtendingExpiration(.never).

I can understand your concerns here. However, I am not feeling good to have different meanings on `.never`: in the `.memoryCacheExpiration` it means "not expiring at all" but in `.memoryCacheAccessExtendingExpiration` now it is "not extending". This would definitely brings confusion and issues in future, even you have written great documentation on it.

Maybe we would need another enum besides of `StorageExpiration` (something like `ExpirationExtending`), or maybe we need to reconsider the name of `.memoryCacheAccessExtendingExpiration` if we want to keep using `StorageExpiration` as the associated value type. I personally prefer the former. Do you have any better idea?

Reply:
OK, we could use `.memoryCacheAccessExtendingExpiration(.seconds(0))` as "not extend" case. I understand your reason to not use `.never` but in such case without special treatment it will change TTL to 0 seconds and expire cache instead of not extending. Of course, we could add there `guard expiration.timeInterval != 0` to change default behavior.

I had something similar to `ExpirationExtending` yesterday, but later removed that. What do you think about such cases:
```swift
enum ExpirationExtending {
case .none
case .cacheTime
case .expirationTime(_ expiration: StorageExpiration)
}```

Default value will be `.cacheTime` as before change.

Reply:
Yes. I think your `ExpirationExtending` definition is so nice!

Reply:
@onevcat I've just pushed the code with `ExpirationExtending`.